mlflow/sklearn/utils.py

```diff
@@ -0,0 +1,186 @@
+from distutils.version import LooseVersion
+import inspect
+from itertools import islice
+
+_SAMPLE_WEIGHT = "sample_weight"
+
+
+def _is_old_version():
+    import sklearn
+
+    return LooseVersion(sklearn.__version__) < LooseVersion("0.20.3")
+
+
+def _all_estimators():
+    import sklearn
+
+    return (
+        sklearn.utils.all_estimators()
+        if hasattr(sklearn.utils, "all_estimators")
+        else _backported_all_estimators()
+    )
+
+
+def _get_Xy(args, kwargs):
+    if len(args) >= 2:
+        return args[:2]
+
+    if len(args) == 1:
+        return args[0], kwargs["y"]
+
+    return kwargs["X"], kwargs["y"]
+
+
+def _get_sample_weight(args_list, args, kwargs):
+    sample_weight_index = args_list.index(_SAMPLE_WEIGHT)
+
+    if len(args) > sample_weight_index:
+        return args[sample_weight_index]
+
+    if _SAMPLE_WEIGHT in kwargs:
+        return kwargs[_SAMPLE_WEIGHT]
+
+    return None
+
+
+def _get_arg_names(f):
+    # `inspect.getargspec` doesn't return a wrapped function's argspec
+    # See: https://hynek.me/articles/decorators/
+    return list(inspect.signature(f).parameters.keys())
+
+
+def _get_args_for_score(fit_func, score_func, args, kwargs):
+    fit_args = _get_arg_names(fit_func)
+    score_args = _get_arg_names(score_func)
+
+    Xy = _get_Xy(args, kwargs)
+
+    if (_SAMPLE_WEIGHT in score_args) and (_SAMPLE_WEIGHT in fit_args):
+        sample_weight = _get_sample_weight(fit_args, args, kwargs)
+        return (*Xy, sample_weight)
+
+    return Xy
+
+
+def _chunk_dict(data, chunk_size=100):
+    # Copied from: https://stackoverflow.com/a/22878842
+
+    it = iter(data)
+    for _ in range(0, len(data), chunk_size):
+        yield {k: data[k] for k in islice(it, chunk_size)}
```

Comment:
A function to chunk a dict by # of keys. I need to create another one to chunk by bytes size.